node/pkg/watchers/evm/finalizers/optimism.go

```diff
@@ -1,95 +1,151 @@
+// Optimism has a Canonical Transaction Chain (CTC) contract running on the L1.
+// This contract contains information about the mapping of L2 => L1 blocks.
+// The Finalizer queries that information and places it in an array.
+// This allows the finalizer to map an L2 block to an L1 block.
+// Then the finalizer can query the L1 chain directly to see if the related L1 block is finalized.
+
+// CTC mainnet contract: 0x5E4e65926BA27467555EB562121fac00D24E9dD2
+// CTC testnet contract: 0x607F755149cFEB3a14E1Dc3A4E2450Cde7dfb04D
+
 package finalizers
 
 import (
 	"context"
 	"fmt"
+	"math/big"
 
 	"github.com/certusone/wormhole/node/pkg/watchers/evm/connectors"
+	ctcAbi "github.com/certusone/wormhole/node/pkg/watchers/evm/finalizers/optimismctcabi"
 	"github.com/certusone/wormhole/node/pkg/watchers/interfaces"
 
+	ethBind "github.com/ethereum/go-ethereum/accounts/abi/bind"
+	ethCommon "github.com/ethereum/go-ethereum/common"
+	ethClient "github.com/ethereum/go-ethereum/ethclient"
+	ethRpc "github.com/ethereum/go-ethereum/rpc"
+
 	"go.uber.org/zap"
 )
 
 // OptimismFinalizer implements the finality check for Optimism.
 // Optimism provides a special "rollup_getInfo" API call to determine the latest L2 (Optimism) block to be published on the L1 (Ethereum).
 // This finalizer polls that API to determine if a block is finalized.
 
-type FinalizerEntry struct {
-	l2Block uint64
-	l1Block uint64
-}
 type OptimismFinalizer struct {
 	logger                 *zap.Logger
 	connector              connectors.Connector
 	l1Finalizer            interfaces.L1Finalizer
-	latestFinalizedL2Block uint64
+	latestFinalizedL2Block *big.Int
```

Comment:
I realized that everywhere in this code base big.Int is being used as a pointer.  So, I made that change here so that this file is consistent with other watcher/finalizer code.